R/RunWorkflow.R

```diff
@@ -30,28 +30,47 @@
 #' @export
 
 RunWorkflow <- function(lWorkflow, lData) {
-  cli::cli_h1(paste0("Initializing `", lWorkflow$meta$file, "` Workflow"))
+  output <- imap(lWorkflow, function(workflow, name){
```

Comment:
Just to clarify this update to RunWorkflow- is the idea here to be able to get `lResults` by just running `RunWorkflow(lWorkflow=wf_metrics, lData=lMapped))` where `wf_metrics` is the list of all workflows for all KRIs? currently we run `lResults <- wf_metrics %>% map(~RunWorkflow(lWorkflow=.x, lData=lMapped))` to do this.

Reply:
yes, that was the idea. I moved the map to the interior, and it will work with one or more workflows, so it will be simpler for the user

Reply:
I do want to refactor RunWorkflow, and I think this is a step in the right direction, but I would like to hold off on the change for now and deal with it in a separate PR. I *really* want to keep the workflow functionality simple. One of the big problems with v1 was that the various workflow functions we created were way too complex. 

A few things to consider: 
- We do want to support running a single workflow (e.g. [mapping](https://github.com/Gilead-BioStats/gsm/blob/4f2b52d4ab453f7ae1b9af9f492504010caa73ea/inst/demo/LongitudinalReport.R#L66)), so we'd need to add logic for that. 
- I want to think through how we use `lResults` vs. `lData` in the object we are returning. If possible, we should just have one of the 2, not both. 
- I want to think about how to "chain" workflows together. Is there any reason we can't call one workflow as a step in a larger workflow? 
- I want to move some of our standard charting and reporting functionality in to workflows as well. Want to make sure updates support that. 

I think we should tackle these changes after the reporting updates are done. It's possible that we'll want to hold off and make the updates as part of v2.1. 

@lauramaxwell @andersonjake1988 for now, let's revert this file to it's previous version and create an issue and PR summarizing the changes needed. We can tackle in the coming weeks.